### PR TITLE
Fix the mandatory fields bug in moderation

### PR DESCRIPTION
**Pull request by SammyIsConfused**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

#1678  
There were a couple other areas in which the mandatory fields check happened - during SaveAndContinue, and during EditInWizard (moderation edits). These were missed during #1708, as the steps to reproduce didn't involve moderation.

A test has been added (VisibilityScriptingBugTest) to verifiy this has been fixed. 
This also includes editing the vanilla institution, which is where the large modified file count comes from. To review just relevant files, add a file filter just allowing .java and .yml files.

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by SammyIsConfused** [comment]
> We now have that one block of logic copy and pasted into three places. Any chance we can centralise this? Maybe a method in a base class - it feels like these might all share one. thinking

I was able to consolidate the EditInWizard and SaveAndContinue methods into one, but the RootWizardSection one had to stay on its own - they don't share a base class.

**Comment by edalex-ian** [comment]
I think we need to optimise that new institution - or can we simply add to an existing. Normally they sit around 1,000 files, but yours is 10x the size at over 10,000 files. Unfortunately that's now in git history, so you're going to want to also do some git surgery I think to cut it out.

File: autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/bugs/VisibilityScriptingBugTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+public class VisibilityScriptingBugTest extends AbstractCleanupTest {
+
+  private static final String METADATA_INPUT_USER = "cpddm";
+  private static final String TOGGLE_USER = "csme";
+  private static final String ADMIN_USER = "TLE_ADMINISTRATOR";
+  private static final String PASS = "tle010";
+  private static final String COLLECTION = "Test Wizard Issue";
+  private static final String ITEM_NAME = "VisibilityScriptingBugTest - Lost Metadata Test Item 1";
+  private static final Date DATE = new Date(1589256742342L);
+  private static final String DATE_AS_STRING = "2020-05-12";
```
Comment (edalex-ian) [inline autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/bugs/VisibilityScriptingBugTest.java:25]: Seeing this is matching `DATE` you should derive it from `DATE` even if that's storing it in a constant.

File: autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/bugs/VisibilityScriptingBugTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  private static final String DROPDOWN = "List 2";
+  private static final String CHECKBOX = "one";
+  private static final String RADIO1 = "Radio 1";
+  private static final String SHUFFLE1 = "Shuffle 1";
+  private static final String SHUFFLE2 = "Shuffle 2";
+  private static final String ROLE = "System Administrator";
+  private static final String RADIO2 = "Yes";
+  private static final String XML_PATH = "item/itembody/";
+  private static final String ROLE_ID = "ROLE_SYSTEM_ADMINISTRATOR";
+  private static final String USER_ID = "4026b550-9a1f-4521-9b56-9ec737c02526";
```
Comment (edalex-ian) [inline autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/bugs/VisibilityScriptingBugTest.java:36]: Good to see the use of constants, but that sure is a lot. Keep in mind you only need them if you're using a value more than once. Also, for a big block like this best to sort them, and if they're related group them into (sorted) blocks. (A good example of this was just trying to find `RADIO2` after I'd already found `RADIO1`.

File: autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/bugs/VisibilityScriptingBugTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import com.tle.webtests.pageobject.viewitem.ItemXmlPage;
+import com.tle.webtests.pageobject.viewitem.SummaryPage;
+import com.tle.webtests.pageobject.wizard.ContributePage;
+import com.tle.webtests.pageobject.wizard.WizardPageTab;
+import com.tle.webtests.test.AbstractCleanupTest;
+import java.util.Date;
+import org.testng.annotations.Test;
+
+@TestInstitution("vanilla")
+public class VisibilityScriptingBugTest extends AbstractCleanupTest {
```
Comment (edalex-ian) [inline autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/bugs/VisibilityScriptingBugTest.java:16]: Be worthwhile in this case to add JavaDoc to the class which has a link off to the GitHub issue.

**Comment by SammyIsConfused** [comment]
As discussed with @edalex-ian, once I get a green build at the current commit I'll apply the changes raised in his review and merge.

**Pull request merged**